lib/src/atom/matcher.rs

```diff
@@ -78,7 +78,7 @@ enum VarResolutionResult<T> {
 /// Abstraction of the variable set. It is used to allow passing both
 /// HashSet<&VariableAtom> and HashSet<VariableAtom> to the
 /// [Bindings::narrow_vars] method.
-pub trait VariableSet {
+pub trait VariableSet : Debug {
```

Comment:
I'm no rust expert but is it normal to keep that?

Reply:
The style-police could come down either way on this.

The Rust stdlib tends to avoid traits bounding other traits when possible and instead puts the bounds on the impls / generics.  But that can get a lot more verbose.

My personal feeling is that the compiler will do dead-code stripping on the whole binary for release, so if the code path isn't needed then it shouldn't bloat the binary.  And knowing trait objects of that type also implement Debug is very convenient and doesn't hurt anything in practice.